### PR TITLE
[svt] fix issue #22897 : change process of video id extraction

### PR DESCRIPTION
**Pull request by uno20001**
## Please follow the guide below

- You will be asked some questions, please read them **carefully** and answer honestly
- Put an `x` into all the boxes [ ] relevant to your *pull request* (like that [x])
- Use *Preview* tab to see how your *pull request* will actually look like

---

### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [adding new extractor tutorial](https://github.com/ytdl-org/youtube-dl#adding-support-for-a-new-site) and [youtube-dl coding conventions](https://github.com/ytdl-org/youtube-dl#youtube-dl-coding-conventions) sections
- [x] [Searched](https://github.com/ytdl-org/youtube-dl/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [x] Checked the code with [flake8](https://pypi.python.org/pypi/flake8)

### In order to be accepted and merged into youtube-dl each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check one of the following options:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [x] Bug fix
- [ ] Improvement
- [ ] New extractor
- [ ] New feature

---

### Description of your *pull request* and other information

This PR fixes issue #22897 by changing the process of extracting video ids.

### CODE REVIEW

File: youtube_dl/extractor/svt.py [hunk 1]
```diff
@@ -360,11 +364,26 @@ def _real_extract(self, url):
 
         webpage = self._download_webpage(url, playlist_id)
 
+        redux_state = self._parse_json(
+            self._search_regex(
+                r'\.\s*reduxState=\s*({.*})',
+                webpage,
+                'reduxState'),
+            None,
+            transform_source=js_to_json)
```
Comment (dstftw) [inline youtube_dl/extractor/svt.py]: Remove superfluous verbosity.

File: youtube_dl/extractor/svt.py [hunk 2]
```diff
@@ -360,11 +364,26 @@ def _real_extract(self, url):
 
         webpage = self._download_webpage(url, playlist_id)
 
+        redux_state = self._parse_json(
+            self._search_regex(
+                r'\.\s*reduxState=\s*({.*})',
```
Comment (dstftw) [inline youtube_dl/extractor/svt.py]: Do not capture empty dict.

File: youtube_dl/extractor/svt.py [hunk 3]
```diff
@@ -360,11 +364,26 @@ def _real_extract(self, url):
...
+                webpage,
+                'reduxState'),
+            None,
```
Comment (dstftw) [inline youtube_dl/extractor/svt.py]: None is not an id.

File: youtube_dl/extractor/svt.py [hunk 4]
```diff
@@ -320,17 +320,24 @@ def _real_extract(self, url):
...
+        },
+        'playlist_count': 2,
     }, {
-        'url': 'https://www.svt.se/nyheter/inrikes/ebba-busch-thor-kd-har-delvis-ratt-om-no-go-zoner',
+        'url': 'https://www.svt.se/nyheter/lokalt/norrbotten/komplicerat-att-stanga-mr-kameran',
         'info_dict': {
-            'id': 'ebba-busch-thor-kd-har-delvis-ratt-om-no-go-zoner',
-            'title': 'Ebba Busch Thor har bara delvis rätt om ”no-go-zoner”',
+            'id': 'komplicerat-att-stanga-mr-kameran',
+            'title': 'MR-kameran fortfarande igång – komplicerad avstängning påbörjad',
```
Comment (dstftw) [inline youtube_dl/extractor/svt.py]: Remove duplicate tests.

File: youtube_dl/extractor/svt.py [hunk 5]
```diff
@@ -360,11 +364,26 @@ def _real_extract(self, url):
...
 
+        redux_state = self._parse_json(
+            self._search_regex(
+                r'\.\s*reduxState=\s*({.*})',
+                webpage,
+                'reduxState'),
+            None,
+            transform_source=js_to_json)
+
+        media = try_get(redux_state, lambda x: x['componentState']['modal']['media']['article']['media'], list) or []
```
Comment (dstftw) [inline youtube_dl/extractor/svt.py]: Inline.

File: youtube_dl/extractor/svt.py [hunk 6]
```diff
@@ -360,11 +364,26 @@ def _real_extract(self, url):
...
+        video_ids = []
+
+        for m in media:
+            if m.get('_type') == 'VIDEOCLIP':
+                video_ids.append(int_or_none(m.get('id')))
+
+        video_ids = filter(lambda x: x is not None, video_ids)
```
Comment (dstftw) [inline youtube_dl/extractor/svt.py]: Move inside a loop.

File: youtube_dl/extractor/svt.py (hunk 4 above)
Comment (uno20001) [reply youtube_dl/extractor/svt.py]: I'm sorry I don't exactly see what you mean. The tests in `SVTPageIE` are all unique.

File: youtube_dl/extractor/svt.py (hunk 1 above)
Comment (uno20001) [reply youtube_dl/extractor/svt.py]: I apologize, but your intent eludes me again, for I do not understand what you mean by *superfluous verbosity*.

File: youtube_dl/extractor/svt.py (hunk 5 above)
Comment (uno20001) [reply youtube_dl/extractor/svt.py]: Could you please clarify what you mean by *inline*?

File: youtube_dl/extractor/svt.py (hunk 2 above)
Comment (uno20001) [reply youtube_dl/extractor/svt.py]: Would a simple `{.+}` suffice as replacement?

File: youtube_dl/extractor/svt.py (hunk 5 above)
Comment (uno20001) [reply youtube_dl/extractor/svt.py]: Upon further investigation of the *coding conventions*, I assume you mean that the *media* variable should be disposed of, is that correct?

**Comment by uno20001** [comment]
I would really appreciate if I could get answers to my questions, because I still have no idea how I should modify the code.

**Pull request closed without merging**